src/viewer.js

```diff
@@ -124,14 +139,47 @@ class Viewer {
   }
   
   /**
-   * @param {?string} urlPath
+   * @param {boolean} isLastBack true if back button was hit and viewer should hide.
+   * @param {number} opt_poppedAmpHistoryIndex The AMP History Index that was just popped.
    * @private
     */
-  handleChangeHistoryState_(urlPath) {
-    if (urlPath) {
-      if (this.showViewer_) this.showViewer_();
-    } else {
-      if (this.hideViewer_) this.hideViewer_();
+  handleChangeHistoryState_(isLastBack, opt_poppedAmpHistoryIndex) {
+    if (this.showViewer_ && this.isViewerHidden_ && this.isViewerHidden_()) {
```

Comment:
This could be true for history events that have nothing to do with the AMP viewer. That is, since the AMP Viewer is embedded on some other page, there could be totally unrelated history events occurring. AMP history needs to correctly ignore then.

I think this code is intended to reopen the viewer on forward. It should verify that the new history state does correspond to the AMP Viewer.